src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -715,7 +735,8 @@ private void RaiseDataReceivedEof()
 
         private unsafe int ProcessRead(SerialStreamIORequest r)
         {
-            Span<byte> buff = r.Buffer.Span;
+            SerialStreamReadRequest readRequest = (SerialStreamReadRequest)r;
```

Comment:
would it be possible to change the type of both queues and avoid the casting?

```diff
- private ConcurrentQueue<SerialStreamIORequest> _readQueue = new ConcurrentQueue<SerialStreamIORequest>();
- private ConcurrentQueue<SerialStreamIORequest> _writeQueue = new ConcurrentQueue<SerialStreamIORequest>();
+ private ConcurrentQueue<SerialStreamReadRequest> _readQueue = new ConcurrentQueue<SerialStreamReadRequest>();
+ private ConcurrentQueue<SerialStreamWriteRequest> _writeQueue = new ConcurrentQueue<SerialStreamWriteRequest>();
```

Reply:
Not really. They get passed to a shared `DoIORequest`:
https://github.com/i3arnon/runtime/blob/851f09c96aaef293b802eb98fd06948c7719f6a3/src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs#L806